elasticsearch/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchQueryProcessor.scala

```diff
@@ -128,8 +128,18 @@ class ElasticSearchQueryProcessor(val logicalPlan: LogicalPlan, val parameters:
   }
 
   private def selectFields(fields: Seq[Attribute], query: SearchDefinition): SearchDefinition = {
-      val stringFields: Seq[String] = fields.map(_.name)
-      query.fields(stringFields.toList: _*)
+      val subDocuments = schemaProvided.toSeq flatMap {
```

Comment:
That's already checked for all native plans at: https://github.com/Stratio/Crossdata/blob/db435cd85aa5eeaf3dbdab0428aec29eefeaa13e/core/src/main/scala/org/apache/spark/sql/crossdata/XDDataFrame.scala#L300